### ISSUE TITLE
[uikits] duplicated pattern menu and broken build with second uikit

### ISSUE THREAD

**Issue by ronalson** [pattern-lab/patternlab-node#901]
I am using Pattern Lab Node `v3.0.0-alpha.16` on `Mac`, with Node `v8.11.1`, using a `Vanilla` Edition with `starterkit-mustache-demo`.

##### Expected Behavior

Using the config bellow and running `npm run pl:build` expected that patterns were built inside `/workshop` directory.

```json
"uikits": [
        {
            "name": "uikit-workshop",
            "outputDir": "workshop",
            "enabled": true,
            "excludedPatternStates": [],
            "excludedTags": []
        }
    ]
```
##### Actual Behavior

Output of `npm run pl:build` appears inside `/workshop` that's right. However, I can also see some files inside `/public`; **Question: is this expected?** 

I tested this with **gulp edition** and the `/public` folder was not filled with any file.

![Screen Shot VSCode](https://user-images.githubusercontent.com/8891651/43020296-ec6ab04c-8c35-11e8-868f-06a71302b10b.png)

### CASE 2: Two Uikits

Add `uikit-bare` following these instructions: https://github.com/pattern-lab/uikit-bare/blob/master/.github/CONTRIBUTING.md

```json
"uikits": [
        {
            "name": "uikit-workshop",
            "outputDir": "workshop",
            "enabled": true,
            "excludedPatternStates": [],
            "excludedTags": []
        },
        {
            "name": "uikit-bare",
            "outputDir": "storefront",
            "enabled": true,
            "excludedPatternStates": [],
            "excludedTags": []
        }
    ]
```
running `npm run pl:build` I have a fill issues:

1. terminal with a bunch of duplicated outputs:

```base
...
Found a lower common denominator pattern state: inreview on templates-detail. Setting reverse lineage pattern pages-contact from <<blank>>
Found a lower common denominator pattern state: inreview on templates-detail. Setting reverse lineage pattern pages-contact from <<blank>>
...
Built Pattern Lab front end
Built Pattern Lab front end
Built Pattern Lab front end
Built Pattern Lab front end
Pattern Lab is watching for changes to files under source/
Pattern Lab is watching for changes to files under source/
Pattern Lab is being served from http://127.0.0.1:3000
Pattern Lab is being served from http://127.0.0.1:3000
Pattern Lab is being served from http://127.0.0.1:3000
Pattern Lab is being served from http://127.0.0.1:3000
```

2. this error (duplicated on terminal)
```bash
{ Error: ENOENT: no such file or directory, open '/xxx/storefront/public/patterns/00-atoms-00-global-00-brand-colors/00-atoms-00-global-00-brand-colors.markup-only.html'
    at Object.fs.openSync (fs.js:646:18)
    at Object.fs.readFileSync (fs.js:551:33)
    at _.each.uikit (xxx/node_modules/@pattern-lab/core/src/lib/markModifiedPatterns.js:48:44)
    at /xxx/node_modules/@pattern-lab/core/node_modules/lodash/lodash.js:4917:15
    at baseForOwn (xxx/node_modules/@pattern-lab/core/node_modules/lodash/lodash.js:3002:24)
    at /xxx/node_modules/@pattern-lab/core/node_modules/lodash/lodash.js:4886:18
    at Function.forEach (xxx/node_modules/@pattern-lab/core/node_modules/lodash/lodash.js:9334:14)
    at cleanPattern (xxx/node_modules/@pattern-lab/core/src/lib/markModifiedPatterns.js:39:7)
    at Array.forEach (<anonymous>)
    at forEachExisting (xxx/node_modules/@pattern-lab/core/src/lib/markModifiedPatterns.js:28:13)
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/xxx/storefront/public/patterns/00-atoms-00-global-00-brand-colors/00-atoms-00-global-00-brand-colors.markup-only.html' }
Error rendering pattern lab header
Error rendering pattern lab header
```
3. the output in the browser has no pattern and duplicated menu items (for both workshop and storefront instances).

![screen shot 2018-07-20 at 16 14 09](https://user-images.githubusercontent.com/8891651/43020938-37ba134c-8c38-11e8-8664-a38fde4ef681.png)

##### Steps to Reproduce

1. Install Pattern-Lab with `edition-node` and `starterkit-mustache-demo` as stated [here](https://github.com/pattern-lab/patternlab-node)
2. Build and Serve with `"name": "uikit-workshop", "outputDir": "workshop",` (CASE 1)
3. Add `uikit-base` as stated [here](https://github.com/pattern-lab/uikit-bare/blob/master/.github/CONTRIBUTING.md)
3. Build and Serve with uikit config from case 2 (CASE 2)

**Comment by ronalson** [comment]
Additional context:

Following those same steps with `edition-gulp` instead of `edition-node`, works as expected.

1. public is clean
2. no duplicated menu
3. patterns are built and shown after `gulp patterlab:serve

> Terminal output shows duplicated messages for every pattern build. But I guess this is ok, right?

```bash
Built pattern: pages-about
Built pattern: pages-about
Built pattern: pages-contact
Built pattern: pages-contact
Built Pattern Lab front end
Built Pattern Lab front end
Pattern Lab is watching for changes to files under source/
Pattern Lab is being served from http://127.0.0.1:3000
Pattern Lab is being served from http://127.0.0.1:3000
http://0.0.0.0:3000 is already in use. Trying another port.
```

**Comment by Hydraner** [comment]
Looks like the reset is happening at the wrong spot

**Comment by JosefBredereck** [comment]
The issue was released with v5.13.0 🎉
